src/main/java/pl/numerki/frontend/ChartGenerator.java

```diff
@@ -0,0 +1,46 @@
+package pl.numerki.frontend;
+
+import org.jfree.chart.ChartFactory;
+import org.jfree.chart.JFreeChart;
+import org.jfree.chart.plot.PlotOrientation;
+import org.jfree.chart.plot.XYPlot;
+import org.jfree.chart.renderer.xy.XYLineAndShapeRenderer;
+import org.jfree.data.xy.XYSeries;
+import org.jfree.data.xy.XYSeriesCollection;
+
+import java.util.function.Function;
+
+public class ChartGenerator {
+    public static JFreeChart generatePlot(
+            Function<Double, Double> function, double leftCompartment, double rightCompartment,
+            double bisectionZeroPosition, double secantZeroPosition
+    ) {
+        XYSeries series = new XYSeries(0);
+        double unit = (rightCompartment - leftCompartment) / 1000;
+        for (double i = leftCompartment; i < rightCompartment; i += unit) {
+            series.add(i, function.apply(i));
+        }
+
+        XYSeries compartmentsAndZeroPosition = new XYSeries(1);
+        compartmentsAndZeroPosition.add(bisectionZeroPosition, 0);
+        compartmentsAndZeroPosition.add(secantZeroPosition, 0);
+
+        XYSeriesCollection seriesCollection = new XYSeriesCollection();
+        seriesCollection.addSeries(series);
+        seriesCollection.addSeries(compartmentsAndZeroPosition);
+
+        JFreeChart chart = ChartFactory.createXYLineChart(
+                "Chart", "x", "y", seriesCollection,
+                PlotOrientation.VERTICAL, false, true, false
+        );
+        XYPlot plot = (XYPlot) chart.getPlot();
+        XYLineAndShapeRenderer renderer = new XYLineAndShapeRenderer();
+        renderer.setSeriesLinesVisible(0, true);
+        renderer.setSeriesShapesVisible(0, false);
+        renderer.setSeriesLinesVisible(1, false);
+        renderer.setSeriesShapesVisible(1, true);
+        plot.setRenderer(renderer);
+
```

Comment:
It would be nice to add additional series x=0 (and y=0 if compartments have opposite signs)
X = 0 Y = 0 should be black :
renderer.setSeriesPaint(0, new Color(0x00, 0X00, 0x00));

Reply:
and also adjust the thickness 
renderer.setSeriesStroke(0, new BasicStroke(4.0f));